File: Contents/InitialConsonantSearch/InitialConsonantSearch/Main/Util/Defines.m
//
//  Defines.m
//  IndicatorSettingsTest
//
//  Created by Kwan Hyun Son on 10/12/23.
//

#import "Defines.h"

@interface Defines ()

@end

@implementation Defines

+ (UIImage * _Nullable)imageNamed:(DefinesImageName)imageName
           imageWithRenderingMode:(UIImageRenderingMode)renderingMode {
    NSBundle *designTimeBundle = [NSBundle bundleForClass:[Defines class]];
    UIImage *result = [UIImage imageNamed:imageName inBundle:designTimeBundle compatibleWithTraitCollection:nil];
    result = [result imageWithRenderingMode:renderingMode];
    return result;
}

@end

File: Contents/InitialConsonantSearch/InitialConsonantSearch_Swift/Main/Defines.swift
//
//  File.swift
//  IndicatorSettingsTest_Swift
//
//  Created by Kwan Hyun Son on 10/12/23.
//

import Foundation


import UIKit

typealias DefinesImageName = String
extension DefinesImageName {
    static let checkmarkCircleFill = "checkmark.circle.fill"
    static let checkmarkCircle = "checkmark.circle"
    static let chevronForward = "chevron.forward"
    static let gearshape = "gearshape"
    static let indent = "indent"
    static let minus = "minus"
    static let plus = "plus"
    static let radioFill = "radio.fill"
    static let radio = "radio"
    static let settingGo = "settingGo"
    static let starFill = "star.fill"
    static let star = "star"
    static let uncheckmarkCircle = "uncheckmark.circle"
    static let uncheckmarkCircleOpaque = "uncheckmark.circle.opaque"
    static let xmark = "xmark"
}

class Defines {
    class func imageNamed(_ imageName: DefinesImageName, renderingMode: UIImage.RenderingMode) -> UIImage? {
        let designTimeBundle = Bundle(for: Defines.self)
        var result = UIImage(named: imageName, in: designTimeBundle, compatibleWith: nil)
        result = result?.withRenderingMode(renderingMode)
        return result
    }
}
